packages/google_maps_flutter/lib/src/marker.dart

```diff
@@ -55,33 +33,135 @@ class InfoWindowText {
+    addIfPresent('anchor', _offsetToJson(anchor));
+
+    return json;
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is InfoWindow &&
+          runtimeType == other.runtimeType &&
+          title == other.title &&
+          snippet == other.snippet &&
+          anchor == other.anchor;
+
+  @override
+  int get hashCode => title.hashCode ^ snippet.hashCode ^ anchor.hashCode;
+
+  @override
+  String toString() {
+    return 'InfoWindow{title: $title, snippet: $snippet, anchor: $anchor}';
+  }
 }
 
-/// Configuration options for [Marker] instances.
+/// Uniquely identifies a [Marker] among [GoogleMap] markers.
 ///
-/// When used to change configuration, null values will be interpreted as
-/// "do not change this configuration option".
-class MarkerOptions {
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class MarkerId {
+  MarkerId(this.value) : assert(value != null);
+
+  /// value of the [MarkerId].
+  final String value;
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is MarkerId &&
+          runtimeType == other.runtimeType &&
+          value == other.value;
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  String toString() {
+    return 'MarkerId{value: $value}';
+  }
+}
+
+/// An icon placed at a particular geographical location on the map's surface.
+/// A marker icon is drawn oriented against the device's screen rather than the
+/// map's surface; that is, it will not necessarily change orientation due to
+/// map rotations, tilting, or zooming.
+@immutable
+class Marker {
   /// Creates a set of marker configuration options.
   ///
-  /// By default, every non-specified field is null, meaning no desire to change
-  /// marker defaults or current configuration.
-  const MarkerOptions({
-    this.alpha,
-    this.anchor,
-    this.consumeTapEvents,
-    this.draggable,
-    this.flat,
-    this.icon,
-    this.infoWindowAnchor,
-    this.infoWindowText,
-    this.position,
-    this.rotation,
-    this.visible,
-    this.zIndex,
+  /// Default marker options.
+  ///
+  /// Specifies a marker that
+  /// * is fully opaque; [alpha] is 1.0
+  /// * uses icon bottom center to indicate map position; [anchor] is (0.5, 1.0)
+  /// * has default tap handling; [consumeTapEvents] is false
+  /// * is stationary; [draggable] is false
+  /// * is drawn against the screen, not the map; [flat] is false
+  /// * has a default icon; [icon] is `BitmapDescriptor.defaultMarker`
+  /// * anchors the info window at top center; [infoWindowAnchor] is (0.5, 0.0)
+  /// * has no info window text; [infoWindowText] is `InfoWindowText.noText`
+  /// * is positioned at 0, 0; [position] is `LatLng(0.0, 0.0)`
+  /// * has an axis-aligned icon; [rotation] is 0.0
+  /// * is visible; [visible] is true
+  /// * is placed at the base of the drawing order; [zIndex] is 0.0
+  const Marker({
```

Comment:
this was previously annotated \@visiblefortesting

Reply:
This is now the preferred way to construct markers. Now that we do not have the `createWithDefaultSettings` factory method

Reply:
ohh right